### DISCUSSION TITLE
Debugger throws System.MissingMethodException, but shows the property correctly inside the object

### DISCUSSION THREAD

**Discussion by Mrcubix** [OmniSharp/omnisharp-vscode#7842] [Q&A]
Made a change to one of my dependencies, adding a property, rebuilt every parts of this C# plugin of mine.
Yet, for some reasons, The debugger refuses to either let me view the value directly, place a condition breakpoint, or a logpoint.

Since this is a plugin, and not a standalone program, I'm remote debugging the Managing app, which loads my plugin.
The PDB files are up to date & the dlls are up to date, as the plugin runs without issues.
Decompilation shows the presence of the property & even the debugger itself shows the property when the containing object is added to watch.

![image](https://github.com/user-attachments/assets/b71b5676-1c4d-47af-8cf0-c2f4b63f409f)

What are the others steps that i should take to make it work normally again?

**Comment by Mrcubix** [comment]
Turns out that yes, my dependency was up to date for this plugin, not another.
The debugger was trying to use the dependency for a complete different plugin, in a complete different context, with a complete different dll & pdb.

So updating that fixed the issue, but i shouldn't have to do that.
If this is by design, the entire design should be reworked, because this is not it.

**Comment by gregg-miskelly** [comment]
To make sure I understand, your complaint is that when you evaluate the property it fails with `'Remote.Driver.IsAttached' a levé une exception de type 'System.MissingMethodException'` ('Remote.Driver.IsAttached' threw an exception of type 'System.MissingMethodException')?

**Reply by gregg-miskelly** [reply]
I should add: Does your process have multiple `Remote.Driver` classes in different assemblies (or the same assembly loaded multiple times)?

**Reply by Mrcubix** [reply]
As I've stated in my reply, there are other plugins running under different contexts, that may have their own copy of the dependency.
It is not loaded in the default context of the application & therefore isn't shared.

So yes, there exist a single `Remote` in multiple Context, but only within one assembly within this context.
You could count that as multiple `Remote` in different assemblies i guess.

```cs
public class Remote
{
    public static RpcClient<IDriverDaemon> Driver { get; } = new("OpenTabletDriver.Daemon");
}
```

**Reply by gregg-miskelly** [reply]
Looking at this some more, the problem couldn't be that 'Remote' exists in multiple assemblies, because the C# EE cannot handle that scenario and you would get an error like: `error CS0433: The type 'Remote' exists in both '<insert-full-name-of-assembly1>' and '<insert-full-name-of-assembly2>`. Instead, it looks like what happens is that the debugger finds the correct method, but it just throws for some reason when the debugger calls it.

**Reply by gregg-miskelly** [reply]
If you add something like the following to your app, and call this method under the debugger, it might provide better information about what is going on:
```
private static bool TestDriver()
{
    try
    {
        return Remote.Driver.IsAttached;
    }
    catch (Exception ex)
    {
        var stackTrace = new StackTrace(ex, false);
        StackFrame[] frames = stackTrace.GetFrames();
        foreach (var frame in frames)
        {
            MethodBase? method = frame.GetMethod();
            if (method != null)
            {
                var module = method.Module;
                var assembly = module.Assembly;
                var moduleName = assembly.Location;
                if (string.IsNullOrEmpty(moduleName) || module.Assembly.ManifestModule != module)
                {
                    moduleName = module.Name;
                }
                Debugger.Log(0, null, $"{moduleName}!{method.DeclaringType?.ToString() ?? "<Global>"}.{method.Name}\n");
            }
        }

        return false;
    }
}
```

**Reply by Mrcubix** [reply]
![image](https://github.com/user-attachments/assets/0ba886e2-1d0e-44eb-b227-407d95e39563)

**Reply by gregg-miskelly** [reply]
Hmm... That is very confusing. Do you by chance have a way to reproduce this problem which is sharable?

**Reply by Mrcubix** [reply]
Probably if i build a project just for this issue, but it would be a lot of work

**Reply by gregg-miskelly** [reply]
Totally understandable, but I don't know what to do with this issue then. I think I will close it for now, unless you have another suggestion.

**Reply by Mrcubix** [reply]
I now have a project that does have an issue, but it's a different one somehow, which still upload it

**Reply by Mrcubix** [reply]
https://github.com/Mrcubix/Debugger-Plugin-Missing-Method

-3 hours that i will never get back

**Reply by gregg-miskelly** [reply]
Thanks for taking the time! 

I am seeing something that could well be the same underlying cause, but is a little different from what you originally reported, so I wanted to make sure I am correctly following your steps before I dig in more. If I hit a breakpoint and evaluate in the Initialize method of `Examples/ExamplePlugin.New/ExampleTool.cs` I see lots of failures, but they are all "Internal error in the C# compiler". Is that what you are seeing also? Or did I do something wrong?

**Reply by Mrcubix** [reply]
This is replicating the same case, but for some reasons, this is also what i'm getting, ~~could either be due to the use of .NET 8 instead of 6?~~ (Nope, same thing with .NET 6) or a related issue. 
This seems like an issue anyway, as i should be able to see values here, in this situation.

**Reply by Mrcubix** [reply]
OH, The Internal C# compiler error is due to the plugins having the exact same assembly Name! @gregg-miskelly 
If you make them different, then you get a more similar issue to what i'm having, i'm pushing the changes to another branch, as the Internal C# compiler error thing is another issue.

**Reply by Mrcubix** [reply]
see https://github.com/Mrcubix/Debugger-Plugin-Missing-Method/tree/FlagC-CS1061 for the more similar issue, this time however, it throws CS1061 instead of the getter method not found.
